.github/workflows/rust.yml

```diff
@@ -14,32 +14,58 @@ env:
 jobs: 
   build:
     runs-on: ubuntu-latest
+    timeout-minutes: 45
     steps:
     - uses: actions/checkout@v3
     - run: cargo build --workspace --verbose
 
   test:
     runs-on: ubuntu-latest
+    timeout-minutes: 45
     steps:
     - uses: actions/checkout@v3
     - run: cargo test --workspace --verbose
 
   clippy:
     runs-on: ubuntu-latest
+    timeout-minutes: 45
     steps:
     - uses: actions/checkout@v3
     - run: cargo clippy --workspace --all-targets --all-features
 
   format:
     runs-on: ubuntu-latest
+    timeout-minutes: 45
     steps:
     - uses: actions/checkout@v3
     - run: cargo fmt --all --check
 
   deny:
     runs-on: ubuntu-latest
+    timeout-minutes: 45
     steps:
     - uses: actions/checkout@v3
     - uses: EmbarkStudios/cargo-deny-action@v1
       with:
         arguments: --all-features --workspace
+
+  docs:
+    runs-on: ubuntu-latest
+    timeout-minutes: 45
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: stable
+        env:
+          RUSTDOCFLAGS: -D warnings
+      - run: cargo doc --workspace --all-features --no-deps --document-private-items
```

Comment:
It would be nice to also publish the Rust docs on GH pages (only main branch), so one can see the pre-release docs somewhere.

Reply:
But that's a nice to have